fragment.go

```diff
@@ -0,0 +1,122 @@
+package vite
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+)
+
+// HTMLFragment generates an HTML fragment for Vite integration based on the provided configuration.
+//
+// This function takes a Config struct and uses it to create the necessary HTML
+// elements for including Vite-managed assets in a web page. The returned HTML
+// is typically intended to be placed in the <head> section of an HTML document.
+//
+// Parameters:
+//   - config: A Config struct containing the necessary configuration options for Vite integration.
+//
+// Returns:
+//   - template.HTML: An HTML fragment as a template.HTML type, which can be safely embedded in HTML templates.
+//   - error: An error if the HTML fragment generation fails for any reason.
+//
+// The returned HTML fragment may include elements such as:
+//   - Script tags for Vite client in development mode
+//   - Links to stylesheets
+//   - Module preload tags
+//   - Script tags for entry points and other necessary JavaScript
+//
+// Usage example:
+//
+//	fragment, err := HTMLFragment(myConfig)
+//	if err != nil {
+//	    // Handle error
+//	}
+//	// Use fragment in your HTML template
+func HTMLFragment(config Config) (template.HTML, error) {
+	pd := &pageData{
+		IsDev:     config.IsDev,
+		ViteEntry: config.ViteEntry,
+		ViteURL:   config.ViteURL,
+	}
+
+	if config.IsDev {
+		pd.PluginReactPreamble = template.HTML(PluginReactPreamble(config.ViteURL))
+	} else {
+		mf, err := config.FS.Open(".vite/manifest.json")
+		if err != nil {
+			return "", fmt.Errorf("vite: open manifest: %w", err)
+		}
+		defer mf.Close()
+
+		m, err := ParseManifest(mf)
+		if err != nil {
+			return "", fmt.Errorf("vite: parse manifest: %w", err)
+		}
+		var chunk *Chunk
+		if chunk == nil {
```

Comment:
This condition is always true.

Reply:
I'll fix.

Reply:
I fixed this issue in this commit [cdc948d44](https://github.com/olivere/vite/pull/10/commits/cdc948d44dba795ce6b23694f6db2e4f0b85e627)